### ISSUE TITLE
Installing Aviation only installs ACM

### ISSUE THREAD

**Issue by Quentin-Despeisse** [ibm-mas/installer#1334]
### MAS CLI version

11.6.0

### CLI function used

install

### What happened?

I choose Aviation as a component for Manage, however on the recap it is ACM that is writtent as Enabled and it doesn't install Aviation. Also, Civil Infrastructure is marked as enabled...

<img width="381" alt="Capture d’écran 2024-11-07 à 14 01 08" src="https://github.com/user-attachments/assets/eb857758-9147-4062-9ce4-0508ada23103">
<img width="467" alt="Capture d’écran 2024-11-07 à 14 01 28" src="https://github.com/user-attachments/assets/39a66e58-58e8-47d5-a5d1-57f4495c58b9">

### Relevant log output

```shell
[ibmmas/cli:11.6.0]mascli$ mas install

IBM Maximo Application Suite Admin CLI v11.6.0
Powered by https://github.com/ibm-mas/ansible-devops/ and https://tekton.dev/


1) Set Target OpenShift Cluster
Already connected to OCP Cluster:
 https://console-openshift-console.apps.672bdf9b9e917720279efee8.ocp.techzone.ibm.com

Proceed with this cluster?? [y/n] y

2) IBM Maximo Operator Catalog Selection
┌─────┬─────────────────┬───────────┬─────────┬──────────┬────────┬──────────┬───────────┬─────────────┬───────────┬──────────────┬────────────┐
│   # │ catalog         │ release   │ core    │ assist   │ iot    │ manage   │ monitor   │ optimizer   │ predict   │ inspection   │ aibroker   │
├─────┼─────────────────┼───────────┼─────────┼──────────┼────────┼──────────┼───────────┼─────────────┼───────────┼──────────────┼────────────┤
│   1 │ v9-241003-amd64 │ 9.0.x     │ 9.0.3   │ 9.0.2    │ 9.0.3  │ 9.0.3    │ 9.0.3     │ 9.0.3       │ 9.0.2     │ 9.0.3        │ 9.0.2      │
├─────┼─────────────────┼───────────┼─────────┼──────────┼────────┼──────────┼───────────┼─────────────┼───────────┼──────────────┼────────────┤
│   2 │ v9-241003-amd64 │ 8.11.x    │ 8.11.15 │ 8.8.6    │ 8.8.13 │ 8.7.12   │ 8.11.11   │ 8.5.9       │ 8.9.5     │ 8.9.6        │            │
├─────┼─────────────────┼───────────┼─────────┼──────────┼────────┼──────────┼───────────┼─────────────┼───────────┼──────────────┼────────────┤
│   3 │ v9-241003-amd64 │ 8.10.x    │ 8.10.18 │ 8.7.7    │ 8.7.17 │ 8.6.18   │ 8.10.14   │ 8.4.10      │ 8.8.3     │ 8.8.4        │            │
├─────┼─────────────────┼───────────┼─────────┼──────────┼────────┼──────────┼───────────┼─────────────┼───────────┼──────────────┼────────────┤
│   4 │ v9-240827-amd64 │ 9.0.x     │ 9.0.2   │ 9.0.2    │ 9.0.2  │ 9.0.2    │ 9.0.2     │ 9.0.2       │ 9.0.1     │ 9.0.2        │            │
├─────┼─────────────────┼───────────┼─────────┼──────────┼────────┼──────────┼───────────┼─────────────┼───────────┼──────────────┼────────────┤
│   5 │ v9-240827-amd64 │ 8.11.x    │ 8.11.14 │ 8.8.6    │ 8.8.12 │ 8.7.11   │ 8.11.10   │ 8.5.8       │ 8.9.3     │ 8.9.5        │            │
├─────┼─────────────────┼───────────┼─────────┼──────────┼────────┼──────────┼───────────┼─────────────┼───────────┼──────────────┼────────────┤
│   6 │ v9-240827-amd64 │ 8.10.x    │ 8.10.17 │ 8.7.7    │ 8.7.16 │ 8.6.17   │ 8.10.13   │ 8.4.9       │ 8.8.3     │ 8.8.4        │            │
├─────┼─────────────────┼───────────┼─────────┼──────────┼────────┼──────────┼───────────┼─────────────┼───────────┼──────────────┼────────────┤
│   7 │ v9-240730-amd64 │ 9.0.x     │ 9.0.1   │ 9.0.1    │ 9.0.1  │ 9.0.1    │ 9.0.1     │ 9.0.1       │ 9.0.0     │ 9.0.0        │            │
├─────┼─────────────────┼───────────┼─────────┼──────────┼────────┼──────────┼───────────┼─────────────┼───────────┼──────────────┼────────────┤
│   8 │ v9-240730-amd64 │ 8.11.x    │ 8.11.13 │ 8.8.5    │ 8.8.11 │ 8.7.10   │ 8.11.9    │ 8.5.7       │ 8.9.3     │ 8.9.4        │            │
├─────┼─────────────────┼───────────┼─────────┼──────────┼────────┼──────────┼───────────┼─────────────┼───────────┼──────────────┼────────────┤
│   9 │ v9-240730-amd64 │ 8.10.x    │ 8.10.16 │ 8.7.6    │ 8.7.15 │ 8.6.16   │ 8.10.12   │ 8.4.8       │ 8.8.3     │ 8.8.4        │            │
└─────┴─────────────────┴───────────┴─────────┴──────────┴────────┴──────────┴───────────┴─────────────┴───────────┴──────────────┴────────────┘
Select catalog and release 1

3) License Terms
To continue with the installation, you must accept the license terms:
 - https://ibm.biz/MAS90-License
 - https://ibm.biz/MaximoIT90-License
 - https://ibm.biz/MAXArcGIS90-License
Do you accept the license terms? [y/n] y

4) Configure Storage Class Usage
Maximo Application Suite and it's dependencies require storage classes that support ReadWriteOnce (RWO) and ReadWriteMany (RWX) access modes:
  - ReadWriteOnce volumes can be mounted as read-write by multiple pods on a single node.
  - ReadWriteMany volumes can be mounted as read-write by multiple pods across many nodes.

Storage provider auto-detected: OpenShift Container Storage
  - Storage class (ReadWriteOnce): ocs-storagecluster-ceph-rbd
  - Storage class (ReadWriteMany): ocs-storagecluster-cephfs
Use the auto-detected storage classes? [y/n] y

5) Configure Product License
License file /mnt/home/license.dat
Contact e-mail address quentindespeisse@ibm.com
Contact first name Quentin
Contact last name Despeisse
IBM Data Reporter Operator (DRO) Namespace redhat-marketplace

6) Configure IBM Container Registry
IBM entitlement key ***********************************************************************************************************************************************************************
*************************

7) Configure MAS Instance
Instance ID restrictions:
 - Must be 3-12 characters long
 - Must only use lowercase letters, numbers, and hypen (-) symbol
 - Must start with a lowercase letter
 - Must end with a lowercase letter or a number
Instance ID mfa

Workspace ID restrictions:
 - Must be 3-12 characters long
 - Must only use lowercase letters and numbers
 - Must start with a lowercase letter
Workspace ID demo

Workspace display name restrictions:
 - Must be 3-300 characters long
Workspace name Demo

8) Configure Operational Mode
Maximo Application Suite can be installed in a non-production mode for internal development and testing, this setting cannot be changed after installation:
 - All applications, add-ons, and solutions have 0 (zero) installation AppPoints in non-production installations.
 - These specifications are also visible in the metrics that are shared with IBM and in the product UI.

  1. Production
  2. Non-Production
Operational Mode 2

9) Certificate Authority Trust
By default, Maximo Application Suite is configured to trust well-known certificate authoritories, you can disable this so that it will only trust the CAs that you explicitly define
Trust default CAs? [y/n] y

10) Cluster Ingress Secret Override
In most OpenShift clusters the installation is able to automatically locate the default ingress certificate, however in some configurations it is necessary to manually configure the name of the secret
Unless you see an error during the ocp-verify stage indicating that the secret can not be determined you do not need to set this and can leave the response empty
Cluster ingress certificate secret name

11) Configure Domain & Certificate Management
Configure domain & certificate management? [y/n] y
Configure custom domain? [y/n] y
MAS top-level domain mfa.ssw-demos.com

DNS Integrations:
  1. Cloudflare
  2. IBM Cloud Internet Services
  3. AWS Route 53
  4. None (I will set up DNS myself)
DNS Provider 1
Cloudflare e-mail quentindespeisse@ibm.com
Cloudflare API token 
Cloudflare zone ssw-demos.com
Cloudflare subdomain mfa
Certificate Issuer:
  1. LetsEncrypt (Production)
  2. LetsEncrypt (Staging)
  3. Self-Signed
Certificate issuer 1
Configure manual certificates? [y/n] n

12) Single Sign-On (SSO)
Many aspects of Maximo Application Suite's Single Sign-On (SSO) can be customized:
 - Idle session automatic logout timer
 - Session, access token, and refresh token timeouts
 - Default identity provider (IDP), and seamless login
 - Brower cookie properties
Configure SSO properties? [y/n] n

13) Configure special characters for userID and username
Do you want to allow special characters for user IDs and usernames?? [y/n] n

14) Enable Guided Tour
By default, Maximo Application Suite is configured with guided tour, you can disable this if it not required
Enable Guided Tour? [y/n] n

15) Application Selection
Install IoT? [y/n] n
Install Manage? [y/n] y
Install Assist? [y/n] n
Install Optimizer? [y/n] n
Install Visual Inspection? [y/n] n
Install AI Broker? [y/n] n

16) Configure Maximo Manage
Customize your Manage installation, refer to the product documentation for more information

16.1) Maximo Manage Components
The default configuration will install Manage with Health enabled, alternatively choose exactly what industry solutions and add-ons will be configured
Select components to enable? [y/n] y
 - Asset Configuration Manager? [y/n] n
 - Aviation? [y/n] y
 - Civil Infrastructure? [y/n] n
 - Envizi? [y/n] n
 - Health? [y/n] n
 - Health, Safety and Environment? [y/n] n
 - Maximo IT? [y/n] n
 - Nuclear? [y/n] n
 - Oil & Gas? [y/n] n
 - Connector for Oracle Applications? [y/n] n
 - Connector for SAP Application? [y/n] n
 - Service Provider? [y/n] n
 - Spatial? [y/n] n
 - Strategize? [y/n] n
 - Transportation? [y/n] n
 - Tririga? [y/n] n
 - Utilities? [y/n] n
 - Workday Applications? [y/n] n

16.2) Maximo Manage Settings - Server Bundles
Define how you want to configure Manage servers:
 - You can have one or multiple Manage servers distributing workload
 - Additionally, you can choose to include JMS server for messaging queues

Configurations:
  1. Deploy the 'all' server pod only (workload is concentrated in just one server pod but consumes less resource)
  2. Deploy the 'all' and 'jms' bundle pods (workload is concentrated in just one server pod and includes jms server)
  3. Deploy the 'mea', 'report', 'ui' and 'cron' bundle pods (workload is distributed across multiple server pods)
  4. Deploy the 'mea', 'report', 'ui', 'cron' and 'jms' bundle pods (workload is distributed across multiple server pods and includes jms server)
Select a server bundle configuration 4
Only Manage JMS sequential queues (sqin and sqout) are enabled by default.
However, you can enable both sequential (sqin and sqout) and continuous queues (cqin and cqout)
Enable both Manage JMS sequential and continuous queues? [y/n] y

16.3) Maximo Manage Settings - Database
Customise the schema, tablespace, indexspace, and encryption settings used by Manage
Customize database settings? [y/n] n

16.4) Maximo Manage Settings - Customization
Provide a customization archive to be used in the Manage build process
Include customization archive? [y/n] n

16.5) Maximo Manage Settings - Other
Configure additional settings:
  - Demo data
  - Base and additional languages
  - Server timezone
  - Cognos integration (install Cloud Pak for Data)
  - Watson Studio Local integration (install Cloud Pak for Data)
Configure Additional Settings? [y/n] y
Create demo data? [y/n] y
Manage server timezone GMT

16.6) Maximo Manage Settings - Languages
Define the base language for Maximo Manage
Base language EN
Define the additional languages to be configured in Maximo Manage. provide a comma-separated list of supported languages codes, for example: 'JA,DE,AR'
A complete list of available language codes is available online:
    https://www.ibm.com/docs/en/mas-cd/mhmpmh-and-p-u/continuous-delivery?topic=deploy-language-support
Secondary languages FR
Integration with Cognos Analytics provides additional support for reporting features in Maximo Manage, for more information refer to the documentation online: 
    https://ibm.biz/BdMuxs
Enable integration with Cognos Analytics? [y/n] n
Enable integration with Watson Studio Local? [y/n] n

17) Configure MongoDb
Install namespace mongoce

18) Configure Databases
The installer can setup one or more IBM Db2 instances in your OpenShift cluster for the use of applications that require a JDBC datasource (IoT, Manage, Monitor, & Predict) or you may choose to configure MAS to use an existing database

18.1) Database Configuration for Maximo Manage
Maximo Manage can be configured to share the system Db2 instance or use it's own dedicated database:
 - Use of a shared instance has a significant footprint reduction but is only recommended for development/test/demo installs
 - In most production systems you will want to use a dedicated database
 - IBM Db2, Oracle Database, & Microsoft SQL Server are all supported database options
Create manage dedicated Db2 instance using the IBM Db2 Universal Operator? [y/n] y
Available Db2 instance types for Manage:
  1. DB2 Warehouse (Default option)
  2. DB2 Online Transactional Processing (OLTP)
Select the Manage dedicated DB2 instance type 1

18.2) Installation Namespace
Install namespace db2u

18.3) Node Affinity and Tolerations
Note that the same settings are applied to both the IoT and Manage Db2 instances
Use existing node labels and taints to control scheduling of the Db2 workload in your cluster
For more information refer to the Red Hat documentation:
 - https://docs.openshift.com/container-platform/4.12/nodes/scheduling/nodes-scheduler-node-affinity.html
 - https://docs.openshift.com/container-platform/4.12/nodes/scheduling/nodes-scheduler-taints-tolerations.html
Configure node affinity? [y/n] n
Configure node tolerations? [y/n] n

18.4) Database CPU & Memory
Note that the same settings are applied to both the IoT and Manage Db2 instances
Customize CPU and memory request/limit? [y/n] n

18.5) Database Storage Capacity
Note that the same settings are applied to both the IoT and Manage Db2 instances
Customize storage capacity? [y/n] n

19) Configure Grafana
Install namespace grafana5
Grafana storage size 10Gi

20) Configure Turbonomic
The IBM Turbonomic hybrid cloud cost optimization platform allows you to eliminate this guesswork with solutions that save time and optimize costs
 - Learn more: https://www.ibm.com/products/turbonomic
Configure IBM Turbonomic integration? [y/n] n

21) Additional Configuration
Additional resource definitions can be applied to the OpenShift Cluster during the MAS configuration step
The primary purpose of this is to apply configuration for Maximo Application Suite itself, but you can use this to deploy ANY additional resource into your cluster
Use additional configurations? [y/n] n

22) Configure Pod Templates
The CLI supports two pod template profiles out of the box that allow you to reconfigure MAS for either a guaranteed or best effort QoS level
For more information about the Kubernetes quality of service (QoS) levels, see https://kubernetes.io/docs/concepts/workloads/pods/pod-qos/
You may also choose to use your own customized pod template definitions
Use pod templates? [y/n] n

23) Non-Interactive Install Command
Save and re-use the following script to re-run this install without needing to answer the interactive prompts again

export IBM_ENTITLEMENT_KEY=x
mas install --mas-catalog-version v9-241003-amd64 --ibm-entitlement-key $IBM_ENTITLEMENT_KEY \
  --mas-channel 9.0.x --mas-instance-id mfa --mas-workspace-id demo --mas-workspace-name "Demo" \
  --non-prod \
  --disable-walkme \
  --storage-class-rwo "ocs-storagecluster-ceph-rbd" --storage-class-rwx "ocs-storagecluster-cephfs" \
  --storage-pipeline "ocs-storagecluster-cephfs" --storage-accessmode "ReadWriteMany" \
  --license-file "/mnt/home/license.dat" \
  --uds-email "quentindespeisse@ibm.com" --uds-firstname "Quentin" --uds-lastname "Despeisse" \
  --dro-namespace "redhat-marketplace" \
  --mongodb-namespace "mongoce" \
  --manage-channel "9.0.x" \
  --manage-jdbc "workspace-application" \
  --manage-components "base=latest,acm=latest" \
  --manage-server-bundle-size "jms" \
  --manage-jms "true" \
  --manage-persistent-volumes \
  --manage-demodata \
  --manage-base-language "EN" \
  --manage-secondary-languages "FR" \
  --manage-server-timezone "GMT" \
  --accept-license --no-confirm

24) Review Settings
Connected to:
 - https://console-openshift-console.apps.672bdf9b9e917720279efee8.ocp.techzone.ibm.com

24.1) OpenShift Container Platform
  Storage Class Provider .................. ocs
  ReadWriteOnce Storage Class ............. ocs-storagecluster-ceph-rbd
  ReadWriteMany Storage Class ............. ocs-storagecluster-cephfs
  Certificate Manager ..................... redhat
  Cluster Ingress Certificate Secret ...... Default
  Single Node OpenShift ................... No
  Skip Pre-Install Healthcheck ............ No
  Skip Grafana-Install .................... No

24.2) IBM Container Registry Credentials
  IBM Entitlement Key ..................... eyJ0eXAi<snip>

24.3) IBM Data Reporter Operator (DRO) Configuration
  Contact e-mail .......................... quentindespeisse@ibm.com
  First name .............................. Quentin
  Last name ............................... Despeisse
  Install Namespace ....................... redhat-marketplace

24.4) IBM Suite License Service
  License File ............................ /mnt/home/license.dat
  IBM Open Registry ....................... icr.io/cpopen

24.5) IBM Maximo Application Suite
  Instance ID ............................. mfa
  Workspace ID ............................ demo
  Workspace Name .......................... Demo

  Operational Mode ........................ Non-Production
  Install Mode ............................ Connected Install

  Domain Name ............................. mfa.ssw-demos.com
  DNS Provider ............................ cloudflare
  Certificate Issuer ...................... mfa-cloudflare-le-prod
  CloudFlare e-mail ....................... quentindespeisse@ibm.com
  CloudFlare API token .................... WsIe_fkn4HvfrDCdINCxtP4MlbNQMRBPPUJ2iZBQ
  CloudFlare zone ......................... ssw-demos.com
  CloudFlare subdomain .................... mfa

  Manual Certificates ..................... None

  Enable Guided Tour ...................... false

  Catalog Version ......................... v9-241003-amd64
  Subscription Channel .................... 9.0.x

  IBM Entitled Registry ................... cp.icr.io/cp
  IBM Open Registry ....................... icr.io/cpopen

  Trust Default Cert Authorities .......... true

  Additional Config ....................... Not Configured
  Pod Templates ........................... Not Configured

24.6) IBM Maximo Application Suite Applications
  IoT ..................................... Do Not Install
  Monitor ................................. Do Not Install
  Manage .................................. 9.0.x
  + Components
    + ACM ................................. Enabled
    + Aviation ............................ Disabled
    + Civil Infrastructure ................ Enabled
    + Envizi .............................. Disabled
    + Health .............................. Disabled
    + HSE ................................. Disabled
    + Maximo IT ........................... Disabled
    + Nuclear ............................. Disabled
    + Oil & Gas ........................... Disabled
    + Connector for Oracle ................ Disabled
    + Connector for SAP ................... Disabled
    + Service Provider .................... Disabled
    + Spatial ............................. Disabled
    + Strategize .......................... Disabled
    + Transportation ...................... Disabled
    + Tririga ............................. Disabled
    + Utilities ........................... Disabled
    + Workday Applications ................ Disabled
  + Server bundle size .................... jms
  + Enable JMS queues ..................... true
  + Server Timezone ....................... GMT
  + Base Language ......................... EN
  + Additional Languages .................. FR
  + Database Settings
    + Schema .............................. Default
    + Username ............................ Default
    + Tablespace .......................... Default
    + Indexspace .......................... Default
  Loc Srv Esri (arcgis) ................... Do Not Install
  Predict ................................. Do Not Install
  Optimizer ............................... Do Not Install
  Assist .................................. Do Not Install
  Visual Inspection ....................... Do Not Install
  AI Broker ............................... Do Not Install

24.7) MongoDb
  Install Namespace ....................... mongoce

24.8) IBM Db2 Univeral Operator Configuration
  System Instance ......................... Do Not Install
  Dedicated Manage Instance ............... Install
   - Type ................................. db2wh
   - Timezone ............................. GMT

  Install Namespace ....................... db2u
  Subscription Channel .................... v110509.0

  CPU Request ............................. 4000m
  CPU Limit ............................... 6000m
  Memory Request .......................... 8Gi
  Memory Limit  ........................... 12Gi

  Meta Storage ............................ 20Gi
  Data Storage ............................ 100Gi
  Backup Storage .......................... 100Gi
  Temp Storage ............................ 100Gi
  Transaction Logs Storage ................ 100Gi

  Node Affinity ........................... None
  Node Tolerations ........................ None

24.9) Cloud Object Storage
  Type .................................... None

24.10) Grafana
  Install Grafana ......................... Install

24.11) Turbonomic
  Turbonomic Integration .................. Disabled

Please carefully review your choices above, correcting mistakes now is much easier than after the install has begun
Proceed with these settings? [y/n] y

25) Launch Install
If you are using storage classes that utilize 'WaitForFirstConsumer' binding mode choose 'No' at the prompt below
Wait for PVCs to bind? [y/n] y
✅️ OpenShift Pipelines Operator is installed and ready to use
✅️ Namespace is ready (mas-mfa-pipelines)
✅️ MAS CLI image deployment test completed
✅️ Latest Tekton definitions are installed (v11.6.0)
✅️ PipelineRun for mfa install submitted
```

**Comment by durera** [comment]
Hah! copy/paste bug, nice easy fix at least:

In the settings:
```
            if self.yesOrNo(" - Aviation"):
                self.params["mas_appws_components"] += ",acm=latest"
```

In the summariser:
```
            self.printSummary("  + Civil Infrastructure", "Enabled" if "acm=" in self.getParam("mas_appws_components") else "Disabled")
```

**Comment by durera** [comment]
Fix will be in v11.8.1, building now so will be available in ~10 minutes or so
